vector/src/main/java/im/vector/app/features/home/room/detail/RoomDetailFragment.kt

```diff
@@ -1556,21 +1557,25 @@ class RoomDetailFragment @Inject constructor(
     }
 
     private fun onShareActionClicked(action: EventSharedAction.Share) {
-        session.fileService().downloadFile(
-                downloadMode = FileService.DownloadMode.FOR_EXTERNAL_SHARE,
-                id = action.eventId,
-                fileName = action.messageContent.body,
-                mimeType = action.messageContent.mimeType,
-                url = action.messageContent.getFileUrl(),
-                elementToDecrypt = action.messageContent.encryptedFileInfo?.toElementToDecrypt(),
-                callback = object : MatrixCallback<File> {
-                    override fun onSuccess(data: File) {
-                        if (isAdded) {
-                            shareMedia(requireContext(), data, getMimeTypeFromUri(requireContext(), data.toUri()))
+        if (action.messageContent is MessageTextContent) {
+            shareText(requireContext(), action.messageContent.body)
```

Comment:
Sadly the body can be totally different than the displayed text, because we can use the formatted_body to render the Event.
Anyway changing that is out of scope of this PR